beer-garden/beer-garden
# Adding back integration testing

This is the test bed for integration testing.

So, this ended up being a lot of tests. So here is what is going on.

Every test spins up the full Beer Garden environment (Beer Garden, MongoDB, and RabbitMQ). We also build Beer-Garden Unstable for this test to make sure we are testing the latest version of the code.

# Tests
Local Plugins:
- Test the example plugins to ensure they turn on and can be interacted with.

Remote Plugins:
- With every combo of Brewtils/Python we support, connect a remote plugin and task it

Garden to Garden
- Connect a Child Garden
- Configure Child Connection
- Verify the Child Systems are on the Parent
- Task Child through Parent
- Task Child through Child
- Verify both requests are on Parent

Overall this takes around 20 minutes to run the full suite because we have to create the docker images.
